### DISCUSSION TITLE
Having each model in separate files and still have just one generated `g.dart`

### DISCUSSION THREAD

**Discussion by rgjurgens** [k-paxian/dart-json-mapper#179] [Q&A] [answered]
Hi, 

In the examples, it looks like for each .dart file where the models are declared in, (with `@jsonSerializable`), you get one generated file along.  But this shouldn’t be the case, right? In the intro it says you have 1 generated file for all model classes.

Also, I'm having issues with reflectable when trying to compile the generated .g file, please see: 

<img width="788" alt="image" src="https://user-images.githubusercontent.com/3368717/154798864-97bf1075-0c76-44fd-81dc-8c0fa0f4d701.png">

Any thoughts on this? I'm an iOS developer originally, so might miss some cases here. 

Thanks! Rudolf

**Comment by k-paxian** [comment]
It seems you are using not the latest package version, since `2.2.1+2` https://pub.dev/packages/dart_json_mapper/changelog#2212
this issue should be resolved.

And yes, there should be only one file generated per each entry point file, in most cases it's only one entry point file like `main.dart`

**Comment by rgjurgens** [comment]
Thanks for you reply! I do have the latest version however, please check my `pubspec.yaml`. 
<img width="601" alt="image" src="https://user-images.githubusercontent.com/3368717/154861418-5e488a40-76fc-41c2-b822-69716dba3455.png">

**Reply by k-paxian** [reply]
Please delete your 'pubspec.lock' and get dependencies once more.

**Reply by rgjurgens** [reply]
Thanks for the help, but I did exactly that and update the dependencies with `flutter pub get` and the problem persists. 
Maybe this screenshot can also help? Does Flutter get the correct version?

<img width="511" alt="Screenshot 2022-02-21 at 11 27 28" src="https://user-images.githubusercontent.com/3368717/154937648-a3a193cb-e832-4798-93a2-9a810ddf3a74.png">

**Reply by k-paxian** [reply]
I see, well the last option left is to specify exact package version. 
```
dart_json_mapper: 2.2.1+3
```

**Comment by rgjurgens** [comment]
This was the issue: 
<img width="788" alt="image" src="https://user-images.githubusercontent.com/3368717/155524646-916dfd92-a0ad-4067-9b8e-2d062cc6f723.png">

**Comment by rgjurgens** [comment]
I'm running into more `path` version conflicts. Also with the `flutter_localizations` library. Any specific reason why you guys chose for this specific `1.8.1` version?

**Reply by k-paxian** [reply]
yep, the reason is: "null safety"

![image](https://user-images.githubusercontent.com/18071010/155984091-bae27d37-55ae-4424-a9a5-08a1261f080b.png)

**Comment by rgjurgens** [comment, marked as answer]
I've overridden the dependency for now 🙏
